File: Samples/azure-storage/Azure.CSharp/Models/CustomDomain.cs

namespace Petstore.Models
{
    using System;
    using System.Linq;
    using System.Collections.Generic;
    using Newtonsoft.Json;
    using Microsoft.Rest;
    using Microsoft.Rest.Serialization;
    using Microsoft.Rest.Azure;

    /// <summary>
    /// The custom domain assigned to this storage account. This can be set
    /// via Update.
    /// </summary>
    public partial class CustomDomain
    {
        /// <summary>
        /// Initializes a new instance of the CustomDomain class.
        /// </summary>
        public CustomDomain() { }

        /// <summary>
        /// Initializes a new instance of the CustomDomain class.
        /// </summary>
        public CustomDomain(string name, bool? useSubDomain = default(bool?))
        {
            Name = name;
            UseSubDomain = useSubDomain;
        }

        /// <summary>
        /// Gets or sets gets or sets the custom domain name. Name is the
        /// CNAME source.
        /// </summary>
        [JsonProperty(PropertyName = "name")]
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets indicates whether indirect CName validation is
        /// enabled. Default value is false. This should only be set on
        /// updates
        /// </summary>
        [JsonProperty(PropertyName = "useSubDomain")]
        public bool? UseSubDomain { get; set; }

        /// <summary>
        /// Validate the object.
        /// </summary>
        /// <exception cref="ValidationException">
        /// Thrown if validation fails
        /// </exception>
        public virtual void Validate()
        {
            if (Name == null)
            {
                throw new ValidationException(ValidationRules.CannotBeNull, "Name");
            }
        }
    }
}

File: Samples/azure-storage/Azure.Java.Fluent/CustomDomain.java
/**
 */

package petstore;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The custom domain assigned to this storage account. This can be set via
 * Update.
 */
public class CustomDomain {
    /**
     * Gets or sets the custom domain name. Name is the CNAME source.
     */
    @JsonProperty(required = true)
    private String name;

    /**
     * Indicates whether indirect CName validation is enabled. Default value
     * is false. This should only be set on updates.
     */
    private Boolean useSubDomain;

    /**
     * Get the name value.
     *
     * @return the name value
     */
    public String name() {
        return this.name;
    }

    /**
     * Set the name value.
     *
     * @param name the name value to set
     * @return the CustomDomain object itself.
     */
    public CustomDomain withName(String name) {
        this.name = name;
        return this;
    }

    /**
     * Get the useSubDomain value.
     *
     * @return the useSubDomain value
     */
    public Boolean useSubDomain() {
        return this.useSubDomain;
    }

    /**
     * Set the useSubDomain value.
     *
     * @param useSubDomain the useSubDomain value to set
     * @return the CustomDomain object itself.
     */
    public CustomDomain withUseSubDomain(Boolean useSubDomain) {
        this.useSubDomain = useSubDomain;
        return this;
    }

}
